### PR TITLE
implement sign in with linkedin backend(make one change to shcema - A…

### PR DESCRIPTION
**Pull request by tra371**
…uthType) WIP* still having issues with post request to linkedin access token endpoint

### CODE REVIEW

File: src/controllers/auth.controller.ts [hunk 1]
```diff
@@ -84,6 +85,98 @@ export const refreshToken = async (
   }
 };
 
+const getOrCreateUser = async (
```
Comment (gvelez17) [inline src/controllers/auth.controller.ts:88]: are we doing this also for github?  

i believe originally i didn't want to create users unless absolutely necessary, but i guess can can, we might want later to create custodial dids for them

should we refactor out the create user part and just make the differences for linkedin and github?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -24,6 +24,7 @@ model User {
 enum AuthType {
   PASSWORD
   GITHUB
+  LINKEDIN
```
Comment (gvelez17) [inline prisma/schema.prisma:27]: ok lets separate out the migration file into its own pr and push that first - maybe you did that already?

File: src/controllers/auth.controller.ts (hunk 1 above)
Comment (tra371) [reply src/controllers/auth.controller.ts:88]: yes we're also doing this for github. matter of fact, i extracted this part from authenticateGithub and made it a separate function.

**Comment by gvelez17** [comment]
@tra371 @0marSalah this would be valuable to have if we can do it carefully
